cmd/tools_gotty.go

```diff
@@ -347,53 +321,25 @@ func gottyAttach(cobraCmd *cobra.Command, o *Options) error { //nolint: funlen
 		o.Global.TopologyFile,
 	)
 
-	// Get lab topology information
-	clabInstance, err := clabcore.NewClabFromTopologyFileOrLabName(
-		o.Global.TopologyFile,
-		o.Global.TopologyName,
-		o.Global.VarsFile,
-		o.Global.Runtime,
-		o.Global.DebugCount > 0,
-		o.Global.Timeout,
-		o.Global.GracefulShutdown,
-	)
+	c, err := clabcore.NewContainerLab(o.ToClabOptions()...)
 	if err != nil {
 		return err
 	}
 
-	labName := clabInstance.Config.Name
-	networkName := clabInstance.Config.Mgmt.Network
-
-	if networkName == "" {
-		networkName = "clab-" + labName
+	rt, ok := c.Runtimes[o.Global.Runtime]
+	if !ok {
+		return fmt.Errorf("failed getting runtime: %w", err)
 	}
 
-	// Set container name if not provided
 	if o.ToolsGoTTY.ContainerName == "" {
-		o.ToolsGoTTY.ContainerName = fmt.Sprintf("clab-%s-gotty", labName)
+		o.ToolsGoTTY.ContainerName = fmt.Sprintf("clab-%s-gotty", c.Config.Name)
 		log.Debugf("Container name not provided, generated name: %s", o.ToolsGoTTY.ContainerName)
 	}
 
-	// Initialize runtime
-	_, rinit, err := clabcore.RuntimeInitializer(o.Global.Runtime)
-	if err != nil {
-		return fmt.Errorf("failed to get runtime initializer for '%s': %w", o.Global.Runtime, err)
-	}
-
-	rt := rinit()
-
-	err = rt.Init(
-		clabruntime.WithConfig(&clabruntime.RuntimeConfig{Timeout: o.Global.Timeout}),
-		clabruntime.WithMgmtNet(&clabtypes.MgmtNet{Network: networkName}),
+	containers, err := c.ListContainers(
+		ctx,
+		clabcore.WithListContainerName(o.ToolsGoTTY.ContainerName),
```

Comment:
add WithListContainerName which I guess only matters for tools but still seems nicer